content/en/docs/concepts/workloads/pods/podpreset.md

```diff
@@ -39,7 +39,9 @@ When a pod creation request occurs, the system does the following:
 1. Attempt to merge the various resources defined by the `PodPreset` into the
    Pod being created.
 1. On error, throw an event documenting the merge error on the pod, and create
-   the pod _without_ any injected resources from the `PodPreset`.
+   the pod _without_ any injected resources from the `PodPreset`. This rule does
```

Comment:
Hi @daixiang0 

This wording is not completely clear. Can you check the grammar?

Reply:
Original info comments that it would succeed in creating pod when error happen. But it is not true.
When use an non-existed secret in podpreset, it would fail to create pod, since the `optional` is not `true`.
If set it as `true`, secret selector would skip check its existence and make pod run successfully as the link says.
All above is what i want to say, i want to simplify but seems more hard to understand.

Reply:
This new wording doesn't seem to be clearly readable as English.
@daixiang0 if you're mainly keen to report that there's an issue, and are happy to let another contributor provide the new working, would you be willing to log an issue about the shortcoming?

Reply:
This part of the documentation is describing what happens if there is an error/conflict merging the resources specified in the podpreset with the existing pod manifest, and is correct as written. It is not describing the behavior when running the pod.

Reply:
@daixiang0 separately, to address your concern, if you specify a secret should be mounted, and do not explicitly mark it as optional, the pod will not start if the secret is not available. That is working as intended.

Reply:
@liggitt I got the key, i misunderstand that the `on error` means any error when create pod, in the other hand, it should be better to change ` Attempt to merge the various resources defined by the `PodPreset` into the Pod being created` to ` Attempt to merge the various resources defined by the `PodPreset` into the Pod before it create`, as you said, all check that podpreset does is before creating pod.